java/code/src/com/redhat/rhn/domain/scc/SCCRepository.java

```diff
@@ -286,25 +286,33 @@ public boolean isAccessible() {
             else {
                 Optional<CredentialsType> currentCredType =
                         Optional.ofNullable(a.getOptionalCredentials().orElse(new Credentials()).getType());
-                if (currentCredType.isPresent() && currentCredType.get().getLabel().equals(Credentials.TYPE_SCC)) {
-                    result = Optional.of(a);
+
+                // if the credentials are of type SCC and have URL that mians that is using a mirror
+                // in this case we want to use it
```

Comment:
This is not correct. If the URL is set, it means "this is the primary credential".
Using a mirror of SCC is not supported. Only "from disk" which has different settings and no credential at all is specified.

Reply:
Thanks for the heads-up. I have changed the comment accordingly. 
A review from you would be much appreciated, especially in this method. I changed it and I think I kept all the previous logic. All pre-existing tests are passing without any change to them, and I added one more to test the newly introduced logic.